locator/src/negative_cache.rs

```diff
@@ -0,0 +1,29 @@
+// Lightweight negative cache which temporarily stores not found results in order to
+// prevent repeated lookups for missing keys.
+use moka::sync::Cache;
+use std::time::Duration;
+
+const SIZE: u64 = 1000;
+const TTL_SECS: u64 = 5;
+
+pub struct NegativeCache {
+    cache: Cache<String, ()>,
+}
+
+impl NegativeCache {
+    pub fn new() -> Self {
+        let cache = Cache::builder()
+            .max_capacity(SIZE)
+            .time_to_live(Duration::from_secs(TTL_SECS))
```

Comment:
Already addresses the two main concerns I would have around staleness and unbounded growth, nice 👍🏻